src/hatch_egg_task.c

```diff
@@ -0,0 +1,107 @@
+#include "hatch_egg_task.h"
+
+#include "get_egg.h"
+#include "naming_screen.h"
+#include "overlay_95.h"
+#include "trainer_memo.h"
+#include "unk_02055244.h"
+#include "unk_020552A4.h"
+
+typedef enum HatchEggTaskState {
+    HATCHEGGTASKSTATE_LEAVE_OVERWORLD,
+    HATCHEGGTASKSTATE_OPEN_HATCH_APP,
+    HATCHEGGTASKSTATE_UPDATE_MON_INFO,
+    HATCHEGGTASKSTATE_OPEN_NAMING_SCREEN,
+    HATCHEGGTASKSTATE_SET_MON_NAME,
+    HATCHEGGTASKSTATE_LOAD_OVERWORLD,
+    HATCHEGGTASKSTATE_EXIT,
+} HatchEggTaskState;
+
+typedef struct HatchEggTaskData {
+    HatchEggTaskState state;
+    BOOL wantToNameNewMon;
+    NamingScreenArgs *namingScreenArgs;
+    UnkStruct_02091240 unkC;
+} HatchEggTaskData;
+
+FS_EXTERN_OVERLAY(OVY_95);
+static const OVY_MGR_TEMPLATE _02106048 = {
```

Comment:
Isn't this the hatch application? or is there still some ambiguity?

Reply:
Missed this one, whoops